core/dictgen/src/rootcling_impl.cxx

```diff
@@ -4519,38 +4519,29 @@ int RootClingMain(int argc,
    // Check if code goes to stdout or rootcling file
    std::ofstream fileout;
    string main_dictname(gOptDictionaryFileName.getValue());
-   std::ostream *dictStreamPtr = NULL;
-   std::ostream *dictStream = nullptr;
+   std::ostream *dictStream = &std::cout;
    std::ostream *splitDictStream = nullptr;
-   std::ostream *splitDictStreamPtr = nullptr;
-   std::unique_ptr<std::ostream> splitDeleter(nullptr);
    // Store the temp files
    tempFileNamesCatalog tmpCatalog;
    if (!gOptIgnoreExistingDict) {
       if (!gOptDictionaryFileName.empty()) {
          tmpCatalog.addFileName(gOptDictionaryFileName.getValue());
          fileout.open(gOptDictionaryFileName.c_str());
-         dictStreamPtr = &fileout;
-         if (!(*dictStreamPtr)) {
+         dictStream = &fileout;
+         if (!(*dictStream)) {
             ROOT::TMetaUtils::Error(0, "rootcling: failed to open %s in main\n",
                                     gOptDictionaryFileName.c_str());
             return 1;
          }
-      } else {
-         dictStreamPtr = &std::cout;
       }
 
       // Now generate a second stream for the split dictionary if it is necessary
       if (gOptSplit) {
-         splitDictStreamPtr = CreateStreamPtrForSplitDict(gOptDictionaryFileName.getValue(), tmpCatalog);
-         splitDeleter.reset(splitDictStreamPtr);
+         splitDictStream = CreateStreamPtrForSplitDict(gOptDictionaryFileName.getValue(), tmpCatalog);
```

Comment:
In the new version, where/when is the ofstream allocated by CreateStreamPtrForSplitDict deleted?

Reply:
Oops, my bad. Thanks for catching it! Should be okay now.